daemon/containerd/progress.go

```diff
@@ -125,38 +126,50 @@ func (p pullProgress) UpdateProgress(ctx context.Context, ongoing *jobs, out pro
 	}
 
 	for _, j := range ongoing.Jobs() {
-		key := remotes.MakeRefKey(ctx, j)
-		if info, ok := pulling[key]; ok {
-			out.WriteProgress(progress.Progress{
-				ID:      stringid.TruncateID(j.Digest.Encoded()),
-				Action:  "Downloading",
-				Current: info.Offset,
-				Total:   info.Total,
-			})
-			continue
-		}
+		switch j.MediaType {
+		case images.MediaTypeDockerSchema2Layer,
+			images.MediaTypeDockerSchema2LayerForeign,
+			images.MediaTypeDockerSchema2LayerGzip,
+			images.MediaTypeDockerSchema2LayerForeignGzip,
+			ocispec.MediaTypeImageLayer,
+			ocispec.MediaTypeImageLayerGzip,
+			ocispec.MediaTypeImageLayerZstd:
```

Comment:
OHMAN, so ugly (not your fault!)

Slightly wondering if this approach would be "brittle" though, with new type of artefacts being added to the spec;

- would there be "other" kind of layers that could exist?
- is there some common denominator that we can use to split "blob" from "manifest"?
- ^^ would the "reverse" work? (i.e., if it's _not_ a manifest)

For the last part, we _do_ have some functions to validate if something is a valid manifest, so wondering if we can piggy-back on that 🤔 

edit; oh! it looks like containerd has utilities for this;

https://github.com/moby/moby/blob/2c95ddf4f3948e5c959f69eaf9ef1eef4d0f6a4d/vendor/github.com/containerd/containerd/images/mediatypes.go#L115-L128


and these (we may have to combine some of them?)

https://github.com/moby/moby/blob/2c95ddf4f3948e5c959f69eaf9ef1eef4d0f6a4d/vendor/github.com/containerd/containerd/images/mediatypes.go#L135-L165